R/descriptives.R

```diff
@@ -31,8 +25,16 @@ DescriptivesInternal <- function(jaspResults, dataset, options) {
     temp <- .descriptivesReadData(options, variables, splitName)
     dataset         <- temp[["dataset"]]
     dataset.factors <- temp[["dataset.factors"]]
-    isNominalText   <- temp[["isNominalText"]]
   }
+  else
+  {
+    dataset.factors <- .readDataSetToEnd(columns.as.factor = c(variables, splitName))
```

Comment:
This works, but wouldn't it make more sense to assume that if `preloadData = TRUE` then `.readDataSetToEnd` is a no-op? This could also be 
```suggestion
    dataset.factors <- dataset
    for (var in c(variables, splitName))
      dataset.factors[[var]] <- as.factor(dataset.factors[[var]])
```
although I guess there is no guarantee that this is _exactly_ the same as if you'd read the variables as factor directly (e.g., read facGender as numeric -> all NA -> as.factor -> all NA vs. read facGender as factor -> actual data). Hmm this really shows descriptives needs another rewrite...

Reply:
well, if you disable preloading in Description.qml it will go back to the old behaviour anyway